pyvista/core/filters/data_set.py

```diff
@@ -318,21 +328,27 @@ def clip_scalar(dataset, scalars=None, invert=True, value=0.0, inplace=False, pr
 
         alg.SetInputDataObject(dataset)
         alg.SetValue(value)
-        if scalars is None:
-            field, scalars = dataset.active_scalars_info
-        _, field = get_array(dataset, scalars, preference='point', info=True)
+        if scalars is not None:
+            dataset.set_active_scalars(scalars)
 
-        # SetInputArrayToProcess(idx, port, connection, field, name)
-        alg.SetInputArrayToProcess(0, 0, 0, field.value, scalars)
         alg.SetInsideOut(invert)  # invert the clip if needed
+        alg.SetGenerateClippedOutput(both)
+
         _update_alg(alg, progress_bar, 'Clipping by a Scalar')
-        result = _get_output(alg)
+        result0 = _get_output(alg)
 
         if inplace:
-            dataset.overwrite(result)
-            return dataset
+            dataset.overwrite(result0)
+            result0 = dataset
+
+        if both:
+            result1 = _get_output(alg, oport=1)
+            if isinstance(dataset, _vtk.vtkPolyData):
+                # For some reason vtkClipPolyData with SetGenerateClippedOutput on leaves unreferenced vertices
+                result0, result1 = (r.clean() for r in (result0, result1))
```

Comment:
Should the result also be cleaned when `both is False`? This may make the changes in #1475 not needed?

Reply:
AFAICT when both is False the filter doesn't return any unreferenced verts, so no need to clean (and doesn't make a difference if #1475 is merged or not)